### PR TITLE
add stream format

### PR DESCRIPTION
**Pull request by mat-hek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,28 @@
...
-    {:membrane_template_plugin, "~> 0.1.0"}
+    {:membrane_stream_format, "~> 0.1.0"}
   ]
 end
 ```
 
-## Usage
-
-TODO
-
```
Comment (mickel8) [inline README.md:27]: What about adding one sentence describing how to use this repo? 

E.g.  You can use this repo for specyfing `caps` in input/output pad e.g. 
```elixir
 def_input_pad :input,
    demand_unit: :buffers,
    caps: [
      {Stream, type: :packet_stream, content: one_of([Opus, nil])}
    ]
```


Or maybe something like in `membrane_opus_format` 
```
Unless you're developing a Membrane Element it's unlikely you need to use this package directly in your app, as normally it is going to be fetched as a dependency of any element that operates on Opus encoded audio.

However, if you are developing an Element or need to add it due to any other reason, the package can be installed by adding membrane_opus_format to your list of dependencies in mix.exs:
```

File: mix.exs [hunk 2]
```diff
@@ -1,12 +1,12 @@
-defmodule Membrane.Template.Mixfile do
+defmodule Membrane.Stream.Format.Mixfile do
```
Comment (mickel8) [inline mix.exs:1]: Cannot add comment to some lines so: we should fix lines 17, 21, 61. There are old `Template` words :slightly_smiling_face:

File: lib/membrane_stream.ex [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  Generic Membrane data stream format.
+
+  Parameters:
+  - content - format that is supposed to be carried in the stream,
+  `nil` if unknown (default)
+  - type - either byte stream (continuous stream) or packet stream (each buffer
+  contains exactly one specified unit of data)
+  """
+  @type t :: %__MODULE__{
+          content: module | nil,
```
Comment (mickel8) [inline lib/membrane_stream.ex:12]: I am wondering if `content` can be missleading here. Someone can think about `content` as some data that is carried in this struct. Maybe change this to `content_format`? On the other hand if someone knows how caps work there shouldn't be any problem with understanding the meaning of this field. What do you think?

**Review by mickel8 — CHANGES_REQUESTED** [review]
We should also change description of repo

File: lib/membrane_stream.ex (hunk 3 above)
Comment (mat-hek) [reply lib/membrane_stream.ex:12]: agreed, content_format seems more intuitive

File: lib/membrane_stream.ex [hunk 4]
```diff
@@ -0,0 +1,16 @@
+defmodule Membrane.Stream do
+  @moduledoc """
+  Generic Membrane data stream format.
```
Comment (bblaszkow06) [inline lib/membrane_stream.ex:3]: TBH, i don't really see the reason for this format to exits. Is there any possible scenario where a format could not be wrapped with Stream? Imagine H264 encoder. Does it produce Stream with content_format H264? Or regular H264? How to make them compatible? Maybe any format should implement behaviour and return `parsed` or `packetized`?
On a second thought, this kind of format might make sense if renamed to `RawStream`, but probably it should be part of core and core should provide some special rules when validating formats to ensure parsed H264 output can be linked to `RawStream(format: H264}` input

Comment (mat-hek) [reply lib/membrane_stream.ex:3]: The intention was to have something like 'raw stream', still not sure if it's a good name, as 'raw' relates to 'unencoded', but it's probably better than just 'stream' anyway. The difference between `Stream{format: H264}` and normal H264 is that the former contains unparsed h264, while the latter - parsed H264. Some elements may accept both, while others may accept only one of them.

Comment (mat-hek) [reply lib/membrane_stream.ex:3]: maybe `StrangerStream`? :P

Comment (mickel8) [reply lib/membrane_stream.ex:3]: Maybe `IncomingStream` or `RemoteStream`

File: README.md [hunk 5]
```diff
@@ -1,34 +1,28 @@
-# Membrane Multimedia Framework: Template Plugin
+# Membrane Remote Stream format
 
-[![Hex.pm](https://img.shields.io/hexpm/v/membrane_template_plugin.svg)](https://hex.pm/packages/membrane_template_plugin)
-[![CircleCI](https://circleci.com/gh/membraneframework/membrane_template_plugin.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_template_plugin)
+[![Hex.pm](https://img.shields.io/hexpm/v/membrane_remote_stream_format.svg)](https://hex.pm/packages/membrane_remote_stream_format)
+[![CircleCI](https://circleci.com/gh/membraneframework/membrane_remote_stream_format.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_remote_stream_format)
```
Comment (mickel8) [inline README.md:4]:
```suggestion
[![Hex.pm](https://img.shields.io/hexpm/v/membrane_remote_stream_format.svg)](https://hex.pm/packages/membrane_remote_stream_format)
[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_remote_stream_format/)
[![CircleCI](https://circleci.com/gh/membraneframework/membrane_remote_stream_format.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_remote_stream_format)
```

**Review by mickel8 — APPROVED** [review]

**Pull request merged**